microprofile/WFLY-12313_MicroProfile_OpenAPI_1.1.adoc

```diff
@@ -0,0 +1,128 @@
+
+* https://thorntail.io/[Thorntail] via the https://github.com/thorntail/thorntail/tree/master/fractions/microprofile/microprofile-openapi[microprofile-openapi fraction].
+** General migration documentation from ThornTail is tracked by https://issues.jboss.org/browse/WFLY-12749[WFLY-12749]
+* https://quarkus.io/[Quarkus] via the https://github.com/quarkusio/quarkus/tree/master/extensions/smallrye-openapi[smallrye-openapi extension].
+
+These integrations will heavily inform, though not dictate, the corresponding integration into WildFly.
+
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Create an MicroProfile OpenAPI galleon layer containing the MP Open API feature encapsulated by a new MicroProfile OpenAPI subsystem.
+** Should depend on MicroProfile Config and JAX-RS (and their dependencies)
+** N.B. New modules to be included in WF:
+*** MicroProfile OpenAPI subsystem module
+*** io.smallrye:smallrye-open-api
+*** org.eclipse.microprofile.openapi:microprofile-openapi-api
+*** com.fasterxml.jackson.dataformat:jackson-dataformat-yaml
+** Which feature packs will use this layer is still TBD.
+* The microprofile-openapi spec module must be exposed to web deployments
+* The `/openapi` endpoint shall be registered relative to the root of the virtual host associated with the deployment.
+** N.B. The `/openapi` endpoint is intended for a public interface, not a management interface, as with the `/health` and `/metrics` endpoints.
+** This places a superficial constraint of a single application per host (see subsequent section re: multiple deployments)
+** In the event a web deployment uses a given host for which another deployment has already registered an `/openapi` endpoint, we will:
+... Log a warning.
+... Skip OpenAPI model generation for that deployment.
+* All WildFly/SmallRye specific configuration will be obtained via MicroProfile Config using the properties prefixed with `mp.openapi.extensions`.
+** Consequently, the microprofile-openapi subsystem should have an empty management model.
+* Support YAML and JSON for both input (static/pre-generated OpenAPI document) and output formats.
+* MicroProfile OpenAPI TCK must pass
+
+=== Nice-to-Have Requirements
+
+* Optional features of the specification:
+** Multiple deployments:
+*** If a container allows multiple deployments, the specification expects the `/openapi` endpoint to return the logical union of the documented services for all applications in the runtime. See: https://download.eclipse.org/microprofile/microprofile-open-api-1.1.2/microprofile-openapi-spec.html#_multiple_applications
+**** Initially, we will limit multiple deployment support to a single deployment per host (for a given endpoint). i.e. each deployment, for which the OpenAPI endpoint is enabled, must either use a distinct host, or a distinct endpoint path (if a given virtual host is associated with multiple deployments).
```

Comment:
As this is still listed among `Nice-to-have`s, it would be good to clarify here what will be the set of requirements - both _hard_ and _nice to have_- that will be delivered in first version of MP integration. This is fundamental information for the user to have a clear idea about if/how the multiple-deployment environment would work.

Reply:
I don't understand what you're asking.  Multiple deployment support is explicitly an optional requirement of the specification, which is why it is listed in the "nice-to-have" requirement section.
I've already described how the initial implementation should handle multiple deployments, and options for future enhancements.  What exactly do you find unclear?

Reply:
Hi @pferraro, probably it's just about terminology: "hard" vs "nice-to-have" requirements let the reader think about this proposal development, not about the specification.
And yes - as a reader - I wanted to be sure that this document states as clearly as possible that initial implementation would have all of the "hard" + not-marked-for-future-enhancements "nice to have"s, and that implementation would work as detailed with respect to multiple deployments. 
Does this makes sense to you?

Reply:
What do you still find unclear about the requirements surrounding multiple deployments?

Reply:
Probably I was just discussing about categories here, i.e. "hard" vs "nice to have" requirements. 
Anyway, I now assume the implementation will follow the hard requirements and those "nice to have"s that are not marked as "Consider for a future enhancement".